script.js

```diff
@@ -1,406 +1,319 @@
-      '1 teaspoon cumin pods, toasted',
-      '1 teaspoon red pepper flakes',
-      '1/2 teaspoon dried oregano',
-      '2 medium cloves garlic, minced (about 2 teaspoons)',
-      '2 tablespoons vegetable or canola oil',
-      '1 whole flap meat steak, 2 to 2 1/2 pounds',
-      'Kosher salt'
-    ],
-    source: 'Serious Eats',
-    totalTime: 240,
-    url: 'http://www.seriouseats.com/recipes/2012/05/spice-rubbed-grilled-flap-meat-sirloin-tip-recipe.html',
-    image: './recipe-images/grilled.jpg'
+];
+
+// basic requirements
+// DOM elements
+const libraryContainer = document.getElementById("library-container");
+const dropdownGenre = document.getElementById("filter-genre");
+const sortButton = document.getElementById("sorting-button");
+const randomBookButton = document.getElementById("random-book");
+const randomBookContainer = document.getElementById("random-book-container");
+
+// display books (with an optional genre filter)
+const showBookContent = (arrayOfBooks, genreFilter = "all") => {
+  // takes an array of book objects and an optional genre filter as parameters
+  libraryContainer.innerHTML = ""; //clears the content of the library container
+
+  // Filter books based on the genre if the filter is not 'all'
+  const filteredBooks =
+    genreFilter === "all"
+      ? arrayOfBooks
+      : arrayOfBooks.filter(
+          (book) => book.genre.toLowerCase() === genreFilter.toLowerCase()
+        );
+
+  //Display filtered books
+  filteredBooks.forEach((book) => {
+    // iterates over each book
+    libraryContainer.innerHTML += `<div class=card>
+     <img id="covers" src=${book.image}>
+     <h2>${book.title}</h2>
+     <p>by ${book.author} | ${book.year}</p>
+     <div class="description">
+      <p>${book.description}</p>
+     </div>
+     <div class="genre-rating">
+      <p>${book.genre}</p>
+      <p>Rating ${book.rating}</p>
+     </div>
+    </div>`;
+  });
+};
+
+// Event listener for dropdown change
+dropdownGenre.addEventListener("change", (event) => {
+  const selectedGenre = event.target.value; // Get the selected genre
+  showBookContent(books, selectedGenre); // Display books based on the selected genre
+});
+
+//call the function - displays all books initially
+showBookContent(books);
+
+//filter by genre
+const filterGenre = () => {};
+
+//sorting function
+let sortAscending = true;
+
+const sortBooksByTitle = () => {
+  books.sort((a, b) => {
+    const titleA = a.title.toUpperCase();
+    const titleB = b.title.toUpperCase();
+    return sortAscending
+      ? titleA.localeCompare(titleB)
+      : titleB.localeCompare(titleA);
+  });
+
+  sortAscending = !sortAscending;
+  showBookContent(books);
+  sortButton.src = sortAscending
+    ? "icons/absteigend-sortieren.png"
+    : "icons/aufsteigend-sortieren.png";
+};
+
+//random button calculations
+const randomBook = () => {
+  const randomIndex = Math.floor(Math.random() * books.length);
+  const selectedBook = books[randomIndex];
+
```

Comment:
Just a suggestion, it will be cool to have this to make sure a random book will not be forever present once it's selected.
```
  // Clear the random book container
  randomBookContainer.innerHTML = "";

  // Display the random book only if both genre and year filters are set to "all"
  if (dropdownGenre.value === "all" && filterByYear.value === "all") {
    randomBookContainer.innerHTML = ` 
      <div class="card random">
        <h2>This is a randomly selected book</h2>
        <img id="covers" src="${selectedBook.image}">
        <h2>${selectedBook.title}</h2>
        <p>by ${selectedBook.author} | ${selectedBook.year}</p>
        <div class="description">
          <p>${selectedBook.description}</p>
        </div>
        <div class="genre-rating">
          <p>${selectedBook.genre}</p>
          <p>Rating ${selectedBook.rating}</p>
        </div>
      </div>`;
  }
```

Reply:
really good suggestion! :)